patches/0093-libavfilter-vulkan-Fix-problem-when-device-have-queu.patch

```diff
@@ -0,0 +1,45 @@
+From ad3d0200c50fcf69e22201db7ba2cd971954e03e Mon Sep 17 00:00:00 2001
+From: Wenbin Chen <wenbin.chen@intel.com>
+Date: Mon, 9 Aug 2021 13:51:31 +0800
+Subject: [PATCH 01/13] libavfilter/vulkan: Fix problem when device have
+ queue_count greater than 1
+
+If the descriptorSetCount is greater than the number of setLayouts,
+vkAllocateDescriptorSets will report error. Now fix it.
+
+Now the following command can run on the device that has queue_count greater
+than one:
+ffmpeg -v verbose -init_hw_device vulkan=vul:0 -filter_hw_device vul
+-i input1080p.264 -vf "hwupload=extra_hw_frames=16,scale_vulkan=1920:1080,
+hwdownload,format=yuv420p" -f rawvideo output.yuv
+
+Signed-off-by: Wenbin Chen <wenbin.chen@intel.com>
+---
+ libavfilter/vulkan.c | 4 ++--
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/libavfilter/vulkan.c b/libavfilter/vulkan.c
+index 337c8d7d5a..e5b070b3e6 100644
+--- a/libavfilter/vulkan.c
++++ b/libavfilter/vulkan.c
+@@ -1160,7 +1160,7 @@ void ff_vk_update_descriptor_set(AVFilterContext *avctx, VulkanPipeline *pl,
+     VulkanFilterContext *s = avctx->priv;
+ 
+     vkUpdateDescriptorSetWithTemplate(s->hwctx->act_dev,
+-                                      pl->desc_set[s->cur_queue_idx * pl->desc_layout_num + set_id],
++                                      pl->desc_set[set_id],
+                                       pl->desc_template[set_id],
+                                       s);
+ }
+@@ -1179,7 +1179,7 @@ int ff_vk_init_pipeline_layout(AVFilterContext *avctx, VulkanPipeline *pl)
+     VkResult ret;
+     VulkanFilterContext *s = avctx->priv;
+ 
+-    pl->descriptor_sets_num = pl->desc_layout_num * s->queue_count;
++    pl->descriptor_sets_num = pl->desc_layout_num;
```

Comment:
I wonder whether pl->descriptor_sets_num is equal to pl->desc_layout_num under any condition. If yes, we may use VulkanPipeline::desc_layout_num only and  remove VulkanPipeline::descriptor_sets_num.

Reply:
Yes, for my patch, they are the same.
According to original code, I think author(Lynn) want to allocate descriptor for each queue to increase performance, and descriptor_sets_num store the total number of descriptor. However the code doesn't work. Now she is aware of this issue and is working on it. I don't know how she want to improve this, so I keep descriptor_sets_num variable.

Reply:
I remove the pl->descriptor_sets_num and use pl->desc_layout_num instead. I double checked the commend from Lynn. She said she will remove all this descriptor and queue management things, because vulkan can do it wisely. I changed the code as what Haihao suggested.